### PR TITLE
crio wipe: add version-file-persist

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
before, crio wipe wasn't precise enough. It always wiped both containers and images on a reboot and version bump. Unfortunately, if the version-file location was somewhere persistent, then no containers would be cleared on reboot.

We should always clear containers on reboot, but should really only clear images on an upgrade. As such, introduce a new flag: --version-file-persist. It should live somewhere persistent, so, if it doesn't exist (or crio is being upgraded, then we can be fairly certain an upgrade happened and it's worth upgrading.

Now, when version-file doesn't exist, but version-file-persist does, we only wipe the containers. If both exist and are the correct version, don't wipe anything. If neither exist, or version-file-persist is out of date, wipe them both

note: we also don't wipe anything when we have temp but not persist. this is to address the update case where we update from an older version of 1.18 that didn't specify a persisted version file. The assumption is if a persisted version file doesn't exist, but the temporary dir does, then we should not update. In the future (when we can more safely assume there will be a persisted version file) this case may mean we should wipe everything.

Also, update the crio wipe tests to test different scenarios

#### Does this PR introduce a user-facing change?

```release-note
add --version-file-persist, a place to put the version file in persistent storage. Now, crio wipe wipes containers if --version-file is not present (presumably it is on temporary storage), and wipes images if both --version-file and --version-file-persist are out of date (presumably there has been an upgrade of cri-o's minor version
```

### CODE REVIEW

File: docs/crio.conf.5.md [hunk 1]
```diff
@@ -45,7 +45,10 @@ CRI-O reads its storage defaults from the containers-storage.conf(5) file locate
   The default log directory where all logs will go unless directly specified by the kubelet. The log directory specified must be an absolute directory.
 
 **version_file**="/var/run/crio/version"
-  Location for crio to lay down the version file.
+  Location for crio to lay down the temporary version file.
+
+**version_file_persist**="/var/lib/crio/version"
+  Location for crio to lay down the persistent version file.
```
Comment (TomSweeneyRedHat) [inline docs/crio.conf.5.md:51]: As noted in the 1.18 review, a little more background would be nice.

**Comment by haircommander** [comment]
merge me plz @umohnani8 @mrunalp

**Comment by haircommander** [comment]
can I have my label back plz @mrunalp @umohnani8

File: test/crio-wipe.bats [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+function teardown() {
+	cleanup_test
+	run_podman_with_args stop -a
+	run_podman_with_args rm -fa
+}
+
+# run crio_wipe calls crio_wipe and tests it succeeded
+function run_crio_wipe() {
+	run $CRIO_BINARY --config "$CRIO_CONFIG" wipe
```
Comment (saschagrunert) [inline test/crio-wipe.bats:31]:
```suggestion
	run $CRIO_BINARY_PATH --config "$CRIO_CONFIG" wipe
```

This :point_up: should fix the static binary CI

Comment (haircommander) [reply test/crio-wipe.bats:31]: AH you're the best thanks

**Pull request merged**